packages/react/src/hooks/useAuth.ts

```diff
@@ -27,32 +23,64 @@ export const useAuth = (): UseAuthResult => {
     user: undefined,
   });
 
-  const handleAuth: HubCallback = ({ payload }) => {
-    switch (payload.event) {
-      case 'signIn':
-        return setResult({ user: payload.data as AuthUser, isLoading: false });
-      case 'signOut':
-        return setResult({ isLoading: false });
-      default:
-        break;
+  /**
+   * Hub events like `tokenRefresh` will not give back the user object.
+   * This util will be used to get current user after those events.
+   */
+  const fetchCurrentUser = React.useCallback(async () => {
+    setResult({ user: undefined, isLoading: true, error: undefined });
+
+    try {
+      const user = (await Auth.currentAuthenticatedUser()) as AmplifyUser;
+      setResult({ user, isLoading: false });
+    } catch (e) {
+      const error = e as Error;
+      setResult({ error, isLoading: false });
     }
-  };
+  }, [setResult]);
```

Comment:
Since `setResult` is memoized, why do we need it here?

Reply:
Very fair, we don't. 8082476d26e4279990b99665cb10b4414fb065a7